UnitsNet.Serialization.JsonNet/UnitsNetJsonConverter.cs

```diff
@@ -184,7 +225,12 @@ public override bool CanConvert(Type objectType)
                 return CanConvertNullable(objectType);
             }
 
-            return objectType.Namespace != null && objectType.Namespace.Equals("UnitsNet");
+            return objectType.Namespace != null && 
+                (objectType.Namespace.Equals("UnitsNet") || 
+                objectType == typeof(ValueUnit) || 
+                // All unit types implement IComparable
+                objectType==typeof(IComparable) ||
+                objectType.FullName.StartsWith("System." + nameof(IComparable)));
```

Comment:
This was very much trial and error and a later refactoring made this unnecessary.